.github/workflows/tests.yaml

```diff
@@ -46,3 +55,8 @@ jobs:
 
       - name: Check resources
         run: kubectl get all -n checkpoint-restore-operator-system
+
+      - name: Wait for Checkpoint Tar Files to Reduce
+        run: |
+          kubectl apply -f ./test/test_checkpointrestoreoperator.yaml
```

Comment:
You could add an `ls -la ${{ env.CHECKPOINT_DIR }}` as a first and last line. Also the `./test/wait_for_checkpoint_reduction.sh` script could print out what it is doing. As it uses `sleep 5` it should be a problem of printing out too much. Print out how many checkpoint it finds in each run of the loop. Something like that. Also try to put a `set -e` into the wait script to ensure we do not skip any possible errors.